src/AlgoliaSearchCore.js

```diff
@@ -609,6 +609,70 @@ AlgoliaSearchCore.prototype.search = function(queries, opts, callback) {
   });
 };
 
+/**
+* Search for facet values
+* https://www.algolia.com/doc/rest-api/search#search-for-facet-values
+* This is the top-level API for SFFV.
+*
+* @param {object[]} queries An array of queries to run.
+* @param {string} queries[].indexName Index name, name of the index to search.
+* @param {object} queries[].params Query parameters.
+* @param {string} queries[].params.facetName Facet name, name of the attribute to search for values in.
+* Must be declared as a facet
+* @param {string} queries[].params.facetQuery Query for the facet search
+* @param {string} [queries[].params.*] Any search parameter of Algolia,
+* see https://www.algolia.com/doc/api-client/javascript/search#search-parameters
+* Pagination is not supported. The page and hitsPerPage parameters will be ignored.
+*/
+AlgoliaSearchCore.prototype.searchForFacetValues = function(queries) {
+  var isArray = require('isarray');
+  var map = require('./map.js');
+
+  var usage = 'Usage: client.searchForFacetValues([{indexName, params}, ...queries])';
+
+  if (!isArray(queries)) {
+    throw new Error(usage);
+  }
+
+  var client = this;
+
+  return Promise.all(map(queries, function performQuery(query) {
+    if (
+      !query ||
+      query.indexName === undefined ||
+      query.params.facetName === undefined ||
+      query.params.facetQuery === undefined
```

Comment:
I'm not following the all picture of this discussion but why do we nest the `facetName`, `facetQuery` and the rest of the parameters in a `params` object? The signature of the function will be a bit less verbose.

```js
client.searchForFacetValues([
  {
    indexName: "products",
    params: {
      facetName: "brand",
      facetValue: "Apple"
      ...searchParameters,
    }
  }
]);

// vs

client.searchForFacetValues([
  {
    indexName: "products",
    facetName: "brand",
    facetValue: "Apple",
    ...searchParameters,
  }
]);
```

Reply:
agreed with the second syntax, we use this kind of pattern in other multi-index methods too

Reply:
That's a good point if it matches the other method signatures.

Reply:
I didn't look correctly to the `search()` signature of the client. It looks the same as the one implemented in the PR.

https://github.com/algolia/algoliasearch-client-javascript/blob/2533c50a9588f5eccff1ebcf9716f476451d9f60/src/AlgoliaSearchCore.js#L564-L578

Reply:
It's the same signature as `search()` indeed. Although I agree `params` is not necessary.